pkg/ebpf/c/tracee.bpf.c

```diff
@@ -970,12 +1033,29 @@ statfunc int walk_mod_tree(program_data_t *p, struct rb_node *root, int idx)
         }
     }
 
-    return HID_MOD_UNCOMPLETED_ITERATIONS;
-}
+    iteration_num++;
 
-struct mod_tree_root {
-    struct latch_tree_root root;
-};
+    if (loop_result == HID_MOD_COMPLETED_ITERATIONS) {
+        flags = HISTORY_SCAN_FINISHED;
+        lkm_seeker_send_to_userspace((struct module *) HISTORY_SCAN_SUCCESSFUL, &flags, &p);
+        bpf_tail_call(ctx, &prog_array, TAIL_HIDDEN_KERNEL_MODULE_PROC);
+    } else if (loop_result == HID_MOD_RACE_CONDITION || iteration_num == MOD_TREE_LOOP_DEPTH) {
+        flags = HISTORY_SCAN_FINISHED;
+        tracee_log(ctx, BPF_LOG_LVL_WARN, BPF_LOG_ID_HID_KER_MOD, loop_result ^ iteration_num);
+        lkm_seeker_send_to_userspace((struct module *) HISTORY_SCAN_FAILURE, &flags, &p);
```

Comment:
Do we really need both a warning and sending the event to userspace?

Reply:
The event must be submitted to userspace to mark that the scan finished.
I could remove the tracee_log and log it in userspace - but it will lose the context of where in the code it happened etc. Passing this information to userspace sounds like an hassle to me where there's the tracee_log API